### ISSUE TITLE
"opening storage failed" error for one prometheus pod

### ISSUE THREAD

**Issue by juzhao** [prometheus/prometheus#7847]
prometheus version: 2.20.0
GCP platform, configured PVs for prometheus
```
apiVersion: v1
data:
  config.yaml: |
    prometheusK8s:
      volumeClaimTemplate:
        metadata:
          name: prometheus
        spec:
          storageClassName: standard
          volumeMode: Filesystem
          resources:
            requests:
              storage: 10Gi
```
the node where prometheus-k8s-1 hosted is NotReady, reboot it, then there is "opening storage failed" error in prometheus-k8s-1 pod, no such issue with prometheus-k8s-0
```
# kubectl -n openshift-monitoring logs -c prometheus prometheus-k8s-1
level=info ts=2020-08-25T08:48:07.950Z caller=main.go:343 msg="Starting Prometheus" version="(version=2.20.0, branch=rhaos-4.6-rhel-7, revision=ba809af96c73832f61371eac19acd8cd94cb3373)"
level=info ts=2020-08-25T08:48:07.950Z caller=main.go:344 build_context="(go=go1.14.4, user=root@5b252a55294c, date=20200821-05:31:40)"
level=info ts=2020-08-25T08:48:07.950Z caller=main.go:345 host_details="(Linux 4.18.0-193.14.3.el8_2.x86_64 #1 SMP Mon Jul 20 15:02:29 UTC 2020 x86_64 prometheus-k8s-1 (none))"
level=info ts=2020-08-25T08:48:07.950Z caller=main.go:346 fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2020-08-25T08:48:07.950Z caller=main.go:347 vm_limits="(soft=unlimited, hard=unlimited)"
level=info ts=2020-08-25T08:48:07.957Z caller=main.go:684 msg="Starting TSDB ..."
level=info ts=2020-08-25T08:48:07.957Z caller=web.go:524 component=web msg="Start listening for connections" address=127.0.0.1:9090
level=info ts=2020-08-25T08:48:07.957Z caller=repair.go:59 component=tsdb msg="Found healthy block" mint=1598325102083 maxt=1598328000000 ulid=01EGJ3WRT97PP17SAPZ6TYQDXB
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:553 msg="Stopping scrape discovery manager..."
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:567 msg="Stopping notify discovery manager..."
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:589 msg="Stopping scrape manager..."
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:563 msg="Notify discovery manager stopped"
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:583 msg="Scrape manager stopped"
level=info ts=2020-08-25T08:48:07.961Z caller=main.go:549 msg="Scrape discovery manager stopped"
level=info ts=2020-08-25T08:48:07.962Z caller=manager.go:888 component="rule manager" msg="Stopping rule manager..."
level=info ts=2020-08-25T08:48:07.962Z caller=manager.go:898 component="rule manager" msg="Rule manager stopped"
level=info ts=2020-08-25T08:48:07.962Z caller=notifier.go:601 component=notifier msg="Stopping notification manager..."
level=info ts=2020-08-25T08:48:07.962Z caller=main.go:755 msg="Notifier manager stopped"
level=error ts=2020-08-25T08:48:07.962Z caller=main.go:764 err="opening storage failed: mmap files, file: /prometheus/chunks_head/000002: mmap: invalid argument"
```

**Comment by roidelapluie** [comment]
Thanks for your report. It looks as if this is actually a question about usage and not development.

It makes more sense to ask questions like this on the [prometheus-users mailing list](https://groups.google.com/forum/#!forum/prometheus-users) rather than in a GitHub issue. On the mailing list, more people are available to potentially respond to your question, and the whole community can benefit from the answers provided.

**Comment by bwplotka** [comment]
Really? This feels like a bug TBH ):

**Comment by roidelapluie** [comment]
> Really? This feels like a bug TBH ):

OK; it seems at first sight unsupported filesystem or problem in the kube storage layer.

**Comment by gouthamve** [comment]
Hi, we've taken a look at this in the bug-scrub and this looks more like a bug. The Prometheus is starting and immediately printing stopping logs. And we need to check if the `00002` file is not an empty file. We should see if we can help make the messaging clear and if the failed TSDB opening is causing the stopping of other components.

**Comment by gouthamve** [comment]
@juzhao Could you check and let us know how big the `/prometheus/chunks_head/000002` file is. does it have data or is it empty?

**Comment by gouthamve** [comment]
Further, this looks like a dupe of https://github.com/prometheus/prometheus/issues/7469. If you find the `000002` to be an empty file, then please delete that file. If its not empty, and then this would be another bug, so please reopen the issue.

**Comment by juzhao** [comment]
> @juzhao Could you check and let us know how big the `/prometheus/chunks_head/000002` file is. does it have data or is it empty?

sorry, the cluster was destroyed, will check the file if we meet it again

**Comment by WoLpH** [comment]
I just ran into the same issue and here's the results for me:
```
/data # cd chunks_head/
/data/chunks_head # ls -lah
total 166M
drwxrwsr-x    2 nobody   nogroup     4.0K Aug 30 04:30 .
drwxrwsr-x   31 root     nogroup     4.0K Aug 29 07:49 ..
-rw-rw-r--    1 nobody   nogroup    77.6M Aug 29 05:00 000508
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 05:00 000509
-rw-rw-r--    1 nobody   nogroup    88.0M Aug 29 06:01 000510
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:03 000511
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:29 000512
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:56 000513
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:10 000514
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:24 000515
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:27 000516
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:29 000517
-rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:30 000518
-rw-rw-r--    1 nobody   nogroup    20.0K Aug 30 04:29 000519
-rw-rw-r--    1 nobody   nogroup        0 Aug 30 04:30 000520
```

**Comment by roidelapluie** [comment]
> I just ran into the same issue and here's the results for me:
> 
> ```
> /data # cd chunks_head/
> /data/chunks_head # ls -lah
> total 166M
> drwxrwsr-x    2 nobody   nogroup     4.0K Aug 30 04:30 .
> drwxrwsr-x   31 root     nogroup     4.0K Aug 29 07:49 ..
> -rw-rw-r--    1 nobody   nogroup    77.6M Aug 29 05:00 000508
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 05:00 000509
> -rw-rw-r--    1 nobody   nogroup    88.0M Aug 29 06:01 000510
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:03 000511
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:29 000512
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 06:56 000513
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:10 000514
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:24 000515
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:27 000516
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:29 000517
> -rw-rw-r--    1 nobody   nogroup        8 Aug 29 07:30 000518
> -rw-rw-r--    1 nobody   nogroup    20.0K Aug 30 04:29 000519
> -rw-rw-r--    1 nobody   nogroup        0 Aug 30 04:30 000520
> ```

Which version of prometheus are you running?

**Comment by WoLpH** [comment]
I'm currently running 2.19.0. But I'm not a 100% certain which version I was running before.

The whole story here:
1. prometheus crashed
2. I did a helm upgrade to see if that would fix it
3. prometheus still crashed, I found out that the disk was too small
4. I resized the disk
5. now prometheus crashed with the error this issue is about
6.. I deleted everything in chunks_head and restarted prometheus again
7. success! it's running :)

I suppose I only should have deleted the 000520 file, but I'm happy it's up and running again. Just very annoying that I had to:
1. kill the prometheus replicaset
2. start a separate debug pod with the pvc mounted
3. manually delete the files
4. delete the debug pod.
5. restart prometheus

It works, but it would be nice if I could exec into the prometheus-server pod directly instead of having to create a pod outside of it.

**Comment by roidelapluie** [comment]
That bug is fixed in 2.19.3, 2.19.0 and 2.20.1

**Comment by WoLpH** [comment]
It seems that the helm chart fixes the version on install, so after the install it's never updated again. Not even with a `helm upgrade` command.

Thanks for the help!
